src/editor/index.jsx

```diff
@@ -36,34 +36,42 @@ import { listenForEvalFramePortReady } from "./port-to-eval-frame";
 
 import "./initialization/initialize-codemirror-loadmode";
 import "./initialization/initialize-dom";
-import { checkNotebookConsistency } from "./actions/actions";
-import { flushServerAutosave } from "./actions/server-actions";
+import { checkNotebookIsBasedOnLatestServerRevision } from "./actions/server-save-actions";
+import { flushServerAutosave } from "./actions/autosave-actions";
+import { restoreLocalAutosave } from "./actions/local-autosave-actions";
+import { connectionModeIsServer } from "./tools/server-tools";
 import CSSCascadeProvider from "../shared/components/css-cascade-provider";
-import { checkForLocalAutosave } from "./tools/local-autosave";
 
 initializeDefaultKeybindings();
 
 window.addEventListener("message", listenForEvalFramePortReady, false);
 
 handleServerVariables(store);
 handleInitialJsmd(store);
-checkForLocalAutosave(store);
+store.dispatch(restoreLocalAutosave());
 handleReportViewModeInitialization(store);
 
 messagePasserEditor.connectDispatch(store.dispatch);
 
 document.addEventListener(
   "visibilitychange",
   () => {
-    if (!document.hidden) {
-      // check notebook consistency if we are returning to this
-      // tab or browser
-      store.dispatch(checkNotebookConsistency());
-    } else {
-      // flush any pending server autosave if we're navigating
-      // away (this will help ensure that a notebook shared
-      // with others e.g. with a copypaste link will be up-to-date)
-      flushServerAutosave();
+    const state = store.getState();
+    if (
+      connectionModeIsServer(state) &&
+      state.userData.name &&
+      state.notebookInfo.user_can_save
+    ) {
+      if (!document.hidden) {
+        // check notebook consistency if we are returning to this
+        // tab or browser
+        store.dispatch(checkNotebookIsBasedOnLatestServerRevision());
+      } else {
+        // flush any pending server autosave if we're navigating
+        // away (this will help ensure that a notebook shared
+        // with others e.g. with a copypaste link will be up-to-date)
+        flushServerAutosave();
+      }
```

Comment:
I was questioning whether this should be a single redux action, though I can't think of an elegant name for it. Thoughts?